libraries/common/list.effekt

```diff
@@ -684,6 +684,25 @@ def isSortedBy[A](list: List[A]) { compare: (A, A) => Bool }: Bool = {
 }
 
 
+
+/// Count how many Elements of a list satisfy the predicate given.
+///
+/// Examples:
+/// ```
+/// >["AB","BC","A","GG","AAA"].countIf{ x => x.length == 2}
+/// 3
+///
+/// >[1 , 2 , 3 , 4].countIf{x => (x*2 > 4)}
+/// 2
+/// ```
+/// O(N)
+def countIf[A](list: List[A]){predicate: (A => Bool) }: Int ={
+  var acc: Int = 0
+  list.foreach {el => if(predicate(el)){acc = acc + 1}}
+  acc
+}
```

Comment:
Please move this function somewhere nearer to `any` and `all` (or other semantically similar functions).
Style-wise, see the previous comments.

Please take a look at similar functions like `partition` to understand the style we're aiming for.

Here's a quick reformat:
```suggestion
/// Count how many elements of a list satisfy the predicate given.
///
/// Examples:
/// ```
/// > ["AB", "BC", "A", "GG", "AAA"].countIf { x => x.length == 2 }
/// 3
///
/// > [1, 2, 3, 4].countIf { x => (x * 2 > 4) }
/// 2
/// ```
///
/// O(N)
def countIf[A](list: List[A]) { predicate: A => Bool }: Int = {
  var acc: Int = 0
  list.foreach { el => 
    if (predicate(el)) {
      acc = acc + 1
    }
  }
  acc
}
```

Reply:
In addition to the above, it would be nice if the tests in `examples/stdlib/list/countIf` were somewhat synchronised with the doc tests presented here. Specifically, I believe the doctests here should be a strict subset of the proper tests in `examples`.